web/packages/selfhosted/test/polyfill/document_embeds/test.ts

```diff
@@ -0,0 +1,50 @@
+import { injectRuffleAndWait, openTest, playAndMonitor } from "../../utils.js";
+import { expect } from "chai";
+
+describe("Document embeds", () => {
+    it("loads the test", async () => {
+        await openTest(browser, `polyfill/document_embeds`);
+    });
+
+    it("Accesses the right number of elements with ruffle", async () => {
+        async function countDocumentEmbeds() {
+            return await browser.execute(() => {
+                return document.embeds?.length ?? 0;
+            });
+        }
+
+        async function documentEmbedsSelfIdentity() {
+            return await browser.execute(() => {
+                return document.embeds === document.embeds;
+            });
+        }
+
+        async function removeEl(selector: string) {
+            const el = await $(selector);
+            await browser.execute((element) => {
+                element.remove();
+            }, el);
+        }
+
+        await injectRuffleAndWait(browser);
+        await playAndMonitor(
+            browser,
+            await browser.$("#test-container").$("ruffle-embed#emb1"),
+        );
+        await playAndMonitor(
+            browser,
+            await browser.$("#test-container").$("ruffle-embed#emb2"),
+        );
+        await playAndMonitor(
+            browser,
+            await browser.$("#test-container").$("ruffle-embed#emb3"),
+        );
+        const documentEmbedsIdentity = await documentEmbedsSelfIdentity();
+        expect(documentEmbedsIdentity).to.equal(true);
+        const embeds1 = await countDocumentEmbeds();
+        expect(embeds1).to.equal(3);
```

Comment:
nit: I think you can inline `embeds1`, `embeds2`, and `documentEmbedsIdentity`, they don't make the code more readable.